src/Token/DataSet.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Token;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ *
+ * @since 4.0.0
+ */
+final class DataSet
+{
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @var string
+     */
+    private $encoded;
+
+    public function __construct(array $data, string $encoded)
+    {
+        $this->data = $data;
```

Comment:
This is super-weird.

 1. Parallel data structure: argh!
 2. No control over its validity